Kernel/Devices/Audio/AC97.cpp

```diff
@@ -274,7 +274,9 @@ ErrorOr<void> AC97::write_single_buffer(UserOrKernelBuffer const& data, size_t o
 void AC97::AC97Channel::handle_dma_stopped()
 {
     dbgln_if(AC97_DEBUG, "AC97 @ {}: channel {}: DMA engine has stopped", m_device.pci_address(), name());
-    VERIFY(m_dma_running);
+    // QEMU might send spurious interrupts while we're not running, so we don't want to panic here.
+    if (!m_dma_running)
+        dbgln("AC97 @ {}: received DMA interrupt while it wasn't running", m_device.pci_address());
     m_dma_running = false;
```

Comment:
Can this spurious interrupt race with the actual enablement of DMA?
If so, we should probably attempt to atomically read `m_dma_running` and avoid setting it to false again. Otherwise, we risk racing and having the spurious interrupt turn it off.

In general, it seems bad that `m_dma_running` isn't handled atomically?

Reply:
I'd suggest to use a small SpinLock scope to ensure we don't have a race condition here as you mentioned. Using an atomic variable is fine if you only need to do the if statement, not if you need to also change it afterwards.